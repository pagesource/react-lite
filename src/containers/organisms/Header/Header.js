// @flow
import get from 'lodash/get';
import { connect } from 'react-redux';
import HeaderComponent from '../../../components/organisms/Header';

const mapStateToProps = state => ({
  nav: get(state, ['global', 'globalData', 'labels', 'header', 'nav']),
});

export default connect(mapStateToProps)(HeaderComponent);
