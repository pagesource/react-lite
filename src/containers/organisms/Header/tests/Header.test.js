import React from 'react';
import { shallow } from 'enzyme';

import MockProvider from '../../../../../__mocks__/mockProvider';

import Header from '../index';

const editorialData = {
  title: 'Test Title',
  banner: 'Test Banner',
};

describe('Header component', () => {
  test('should render correctly', () => {
    const HeaderComponent = shallow(
      <MockProvider>
        <Header editorialData={editorialData} />
      </MockProvider>
    );
    expect(HeaderComponent).toMatchSnapshot();
  });
});
