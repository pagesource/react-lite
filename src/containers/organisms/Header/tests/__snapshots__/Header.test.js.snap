// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Connect(Header)
      editorialData={
        Object {
          "banner": "Test Banner",
          "title": "Test Title",
        }
      }
    />
  </MemoryRouter>
</Provider>
`;
